version.py

```diff
@@ -27,14 +27,69 @@
 """
 import os
 import re
+import argparse
+import logging
+import subprocess
 
 # current version
 # We use the version of the incoming release for code
 # that is under development
 __version__ = "0.8.dev0"
+# most recent tag, used for git describe validation
+__most_recent_tag__ = "v0.7.0"
```

Comment:
minor suggestion: given `__version__` and `__most_recent_tag__` are constants, wouldn't make sense to spell them as `VERSION` and `MOST_RECENT_TAG` instead?

Reply:
I think we can keep the style in this case to be consistent with the rest of the code base where version is used. Since version global variable is somewhat different from other global constants

Reply:
Ack.